e2e-tests/portal/cypress/e2e/journeys/login/navigating-using-sign-in-link.spec.js

```diff
@@ -0,0 +1,143 @@
+      cy.enterUsernameAndPassword(BANK1_MAKER1);
+    });
+
+    it('Opening a previously issued sign in link takes the user to the /login/sign-in-link-expired page to resned link and does not give the user access to protected routes', () => {
+      cy.overridePortalUserSignInTokensByUsername({ username: BANK1_MAKER1.username, newSignInTokens: [EXPIRED_SIGN_IN_TOKEN, NOT_EXPIRED_SIGN_IN_TOKEN] });
+      signInLink.visit({ token: EXPIRED_SIGN_IN_TOKEN.signInTokenFromLink, userId: bank1Maker1Id });
+
+      linkShouldBeExpiredAndAbleToBeResent();
+
+      checkUserDoesNotHaveAccessToProtectedRoutes();
+    });
+
+    it('Opening a previously issued but not expired sign in link takes the user to the /login/sign-in-link-expired page to resned link and does not give the user access to protected routes', () => {
+      cy.overridePortalUserSignInTokensByUsername({ username: BANK1_MAKER1.username, newSignInTokens: [ANOTHER_NOT_EXPIRED_TOKEN, NOT_EXPIRED_SIGN_IN_TOKEN] });
+      signInLink.visit({ token: ANOTHER_NOT_EXPIRED_TOKEN.signInTokenFromLink, userId: bank1Maker1Id });
+
+      linkShouldBeExpiredAndAbleToBeResent();
+
+      checkUserDoesNotHaveAccessToProtectedRoutes();
+    });
+
+    it('Opening the most recently issued, but expired, sign in link takes the user to the /login/sign-in-link-expired page to resned link and does not give the user access to protected routes', () => {
+      cy.overridePortalUserSignInTokensByUsername({ username: BANK1_MAKER1.username, newSignInTokens: [EXPIRED_SIGN_IN_TOKEN] });
+      signInLink.visit({ token: EXPIRED_SIGN_IN_TOKEN.signInTokenFromLink, userId: bank1Maker1Id });
+
+      linkShouldBeExpiredAndAbleToBeResent();
+
+      checkUserDoesNotHaveAccessToProtectedRoutes();
+    });
+
+    it('Opening an invalid sign in link takes the user to the page not found page and does not give the user access to protected routes', () => {
+      cy.overridePortalUserSignInTokensByUsername({ username: BANK1_MAKER1.username, newSignInTokens: [INVALID_SIGN_IN_TOKEN] });
+      signInLink.visit({ token: INVALID_SIGN_IN_TOKEN.signInTokenFromLink, userId: bank1Maker1Id }, { failOnStatusCode: false });
+      signInLink.shouldDisplayProblemWithServiceError();
+
+      checkUserDoesNotHaveAccessToProtectedRoutes();
+    });
+
+    it('Opening a valid sign in link takes the user to the /dashboard page and gives the user access to protected routes', () => {
+      cy.overridePortalUserSignInTokensByUsername({ username: BANK1_MAKER1.username, newSignInTokens: [EXPIRED_SIGN_IN_TOKEN, NOT_EXPIRED_SIGN_IN_TOKEN] });
+
+      signInLink.visit({ token: NOT_EXPIRED_SIGN_IN_TOKEN.signInTokenFromLink, userId: bank1Maker1Id });
+      cy.url().should('eq', relative('/dashboard/deals/0'));
+
+      beforeYouStart.visit();
+      cy.url().should('eq', relative('/before-you-start'));
+    });
+
+    it('Opening a valid sign in link when user is blocked takes the user to the user blocked page', () => {
```

Comment:
Ah, great catch. I'll add this in quickly instead of getting a new ticket for it

Reply:
We do currently still check this in login controller -- We can address this after the patch if @abhi-markan is okay with this. The only way this would occur is if you blocked a user and they had got to the "send 2fa link" screen and their user&pass token hasnt expired (30 mins).